runtime/src/iree/hal/drivers/cuda/nccl_channel.c

```diff
@@ -122,7 +130,22 @@ static void iree_hal_cuda_nccl_channel_destroy(
   //  syms->ncclCommDestroy(channel->comm)
   // Should work the same (as we are doing a blocking teardown):
   //  syms->ncclCommDestroy(channel->comm)
-
+  NCCL_IGNORE_ERROR(channel->context_wrapper->syms,
+                    ncclCommFinalize(channel->comm));
+  ncclResult_t async_error;
+  for (;;) {
+    NCCL_IGNORE_ERROR(channel->context_wrapper->syms,
+                      ncclCommGetAsyncError(channel->comm, &async_error));
+    if (async_error == ncclInProgress) {
+      // wait for a second
+      const iree_time_t sec_in_ns = 1000000000;
+      iree_wait_until(sec_in_ns);
```

Comment:
Isn't this wait too long?

Reply:
I don't have a clue whether it is very long or not. Please suggests.

Reply:
I have removed this code since the current channel creation always uses the blocking mode.